catalog/app/utils/Config.js

```diff
@@ -18,32 +18,37 @@ const parseJSON = (msg = 'invalid JSON') =>
     throw new ConfigError(`${msg}:\n${src}`, { src, originalError: e })
   })
 
+const SSO_PROVIDERS = ['google']
+
 const validateConfig = conforms({
   registryUrl: R.is(String),
   alwaysRequiresAuth: R.is(Boolean),
-  sentryDSN: isNullable(String),
-  intercomAppId: isNullable(String),
-  mixpanelToken: isNullable(String),
+  sentryDSN: isNullable(R.is(String)),
+  intercomAppId: isNullable(R.is(String)),
+  mixpanelToken: isNullable(R.is(String)),
   apiGatewayEndpoint: R.is(String),
   defaultBucket: R.is(String),
   signInRedirect: R.is(String),
   signOutRedirect: R.is(String),
-  disableSignUp: isNullable(Boolean),
+  passwordAuth: isNullable(oneOf([true, false, 'SIGN_IN_ONLY'])),
```

Comment:
@akarve: we won't be able to do JSON validation once we add any of the new parameters here - or at least, I can't think of a way to do it.

The new `config.json`, before CFT substitutes whatever parameters, will need to be something like this:
```
  ...
  "passwordAuth": ${PasswordAuth},
  "ssoAuth": ${SsoAuth},
  "ssoProviders": ${SsoProviders},
   ...
```
But, that's no longer valid JSON. We _could_ switch from booleans and arrays to strings... but that's kind of ugly.

Reply:
I think we should add json-schema overrides in `/deployment` (and validate after the overrides are applied to the json-schema) so that some keys can be validated as "Sub expressions" at template time. But the catalog should evaluate the final, JSON form of the strong. We probably will have to turn some booleans into strings, though, since "Sub expressions" can't emit JSON. I think that's a good compromise.